src/shared/cfi.ts

```diff
@@ -34,6 +34,28 @@ function compareArrays(
   return a.length - b.length;
 }
 
+/**
+ * Split a hyphen-separated CFI range.
+ *
+ * CFI assertions are stripped in the process. If `range` does not contain a
+ * hyphen, the result will be an empty range with the end point being the same
+ * as the start point.
+ *
+ * @example
+ *   splitCFIRange("/2/4[chap-02]-/2/6[chap-03]") // returns `["/2/4", "/2/6"]`.
+ */
+export function splitCFIRange(range: string): [string, string] {
+  const rangeWithoutAssertions = stripCFIAssertions(range);
+
+  // eslint-disable-next-line prefer-const
+  let [start, end] = rangeWithoutAssertions.split('-', 2);
+  if (end === undefined) {
+    end = start;
+  }
+
+  return [start, end];
```

Comment:
You could use `const` here by setting `start` as the default value for `end`:

```suggestion
  const [start, end = start] = rangeWithoutAssertions.split('-', 2);
  return [start, end];
```

Reply:
Alternatively, you can do `const [start, end = start] = ...;` and `return [start, end ?? start];` if you think it's more clear.

Reply:
The second option sounds good. I'll do that.